C#: Rework linters/analyzers approach (#3252)
valkey-io/valkey-glide: csharp/DEVELOPER.md

dotnet test
```

> You can also specify which framework version to use for testing (by defaults it runs on net6.0 and net8.0) by adding `--framework net8.0` or `--framework net6.0` accordingly.

Author: Right. We have time to make a desicion before release.
That C# client draft exists for about 2 years, things changed since that.

> You can also specify which framework version to use for testing (by defaults it runs on net6.0 and net8.0) by adding `--framework net8.0` or `--framework net6.0` accordingly.
> By default, `dotnet test` produces no reporting and no logging for tests, to define a logger add `--logger "html;LogFileName=TestReport.html"` (HTML reporting) or `--logger "console;verbosity=detailed"` (console reporting) or both.
> To filter test by class name or method name add the following expression to the command line: `--filter "FullyQualifiedName~<test or class name>"` ([see more in docs](https://learn.microsoft.com/en-us/dotnet/core/testing/selective-unit-tests?pivots=xunit)).
> A command line may contain all listed above parameters, for example:

Reviewer: ```suggestion
You can also specify which framework version to use for testing (by defaults it runs on net6.0 and net8.0) by adding `--framework net8.0` or `--framework net6.0` accordingly.

By default, `dotnet test` produces no reporting and does not display the test results.  To log the test results to the console and/or produce a test report, you can use the `--logger` attribute with the test command.  For example:

- `dotnet test --logger "html;LogFileName=TestReport.html"` (HTML reporting) or
- `dotnet test --logger "console;verbosity=detailed"` (console reporting)

To filter tests by class name or method name add the following expression to the command line: `--filter "FullyQualifiedName~<test or class name>"` (see the [.net testing documentation](https://learn.microsoft.com/en-us/dotnet/core/testing/selective-unit-tests?pivots=xunit) for more details).

A command line may contain all listed above parameters, for example:
```

5. Lint the code

> Before making a contribution ensure that all new user API and non-obvious places in code is well documented and run code linters and analyzers.

Reviewer 2: For API we could even enforce it if we feel it makes sense. I'm not sure we need to though.

> Before making a contribution ensure that all new user API and non-obvious places in code is well documented and run code linters and analyzers.

Reviewer: ```suggestion
Before making a contribution, ensure that all new user APIs and non-obvious code is well documented, and run the code linters and analyzers.
```

6. Test framework and style

> Test package used in code xUnit v3. Testing code styles defined in `.editorcofing` (see `dotnet_diagnostic.xUnit..` rules). Rules enforced by https://github.com/xunit/xunit.analyzers referenced by the main xunit.v3 NuGet package out of the box. If you choose to reference xunit.v3.core instead, you can reference xunit.analyzers explicitly. For additional info, please, refer to https://xunit.net and https://github.com/xunit/xunit

Reviewer: ```suggestion
The CSharp Valkey-Glide client uses xUnit v3 for testing code. The test code styles are defined in `.editorcofing` (see `dotnet_diagnostic.xUnit..` rules). The xUnit rules are enforced by the [xUnit analyzers](https://github.com/xunit/xunit.analyzers) referenced in the main xunit.v3 NuGet package. If you choose to use xunit.v3.core instead, you can reference xunit.analyzers explicitly. For additional info, please, refer to https://xunit.net and https://github.com/xunit/xunit
```
